File: streamingpro-spark-2.2.0-adaptor/src/main/java/org/apache/spark/MLSQLSyntaxErrorListener.scala
package org.apache.spark

import org.antlr.v4.runtime.atn.ATNSimulator
import org.antlr.v4.runtime.{BaseErrorListener, RecognitionException, Recognizer}
import org.apache.spark.internal.Logging

class MLSQLSyntaxErrorListener extends BaseErrorListener with Logging {
  override def syntaxError(recognizer: Recognizer[_, _],
                           offendingSymbol:
                           scala.Any,
                           line: Int,
                           charPositionInLine: Int,
                           msg: String,
                           e: RecognitionException): Unit = {
    throw new RuntimeException("MLSQL Parser error in [row:" + line + " column:" + charPositionInLine + "]  error msg: " + msg)
  }
}

File: streamingpro-spark-2.3.0-adaptor/src/main/java/org/apache/spark/MLSQLSyntaxErrorListener.java
package org.apache.spark;

import org.antlr.v4.runtime.BaseErrorListener;
import org.antlr.v4.runtime.RecognitionException;
import org.antlr.v4.runtime.Recognizer;

public class MLSQLSyntaxErrorListener extends BaseErrorListener {

    @Override
    public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine, String msg, RecognitionException e) {
        throw new RuntimeException("MLSQL Parser error in [row:" + line + " column:" + charPositionInLine + "]  error msg: " + msg);
    }
}
